src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/InputOverloadMethodsDeclarationOrderRecords.java

```diff
@@ -21,7 +21,7 @@ public void notFoo() {
         public void foo(int i, String s) { // violation
         }
 
-        public void foo(String s, int i) {
+        public void foo(String s, int i) { // violation
```

Comment:
I'm wondering why this wasn't marked as a violation before (along with lines 63, 66)? By rule this should be next to `foo` on line 15. If user only fixes violation for line 21, the next check they will get a violation for this method. Without this PR, it's probably obvious to user what methods need to be grouped together. However, if / when this PR is merged, complex regex patterns can make it not as obvious where a method should be put after. It might become burdensome for users to have to repeatedly run the checks as they fix violations one-by-one instead of seeing them all at once.  I can easily revert to old behavior, but IMO I think this is correct

Reply:
> I can easily revert to old behavior, but IMO I think this is correct

If this is the case, can we do this in the scope of a separate issue?